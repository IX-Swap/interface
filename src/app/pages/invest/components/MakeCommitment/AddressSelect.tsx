import React from 'react'
import { useWithdrawalAddresses } from 'app/pages/accounts/pages/withdrawalAddresses/hooks/useWithdrawalAddresses'
import { TypedField } from 'components/form/TypedField'
import { WithdrawalAddressSelect } from 'components/form/WithdrawalAddressSelect'
import { useFormContext } from 'react-hook-form'

export const AddressSelect = () => {
  const { control } = useFormContext()
  const { data, status } = useWithdrawalAddresses({})
  const filteredAddresses = data.list.filter(
    ({ status }) => status === 'Approved'
  )
  const hasFilteredAddresses = filteredAddresses.length > 0

  return (
    <TypedField
      component={WithdrawalAddressSelect}
      control={control}
      name='withdrawalAddress'
      label='Destination Wallet Address'
      placeholder={
        !hasFilteredAddresses && status !== 'loading'
          ? 'You do not have wallet addresses'
          : 'Select walet'
      }
      disabled={!hasFilteredAddresses}
      displayEmpty
      list={filteredAddresses}
      status={status}
    />
  )
}
