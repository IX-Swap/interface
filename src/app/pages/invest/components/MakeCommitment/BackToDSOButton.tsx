import React from 'react'
import { Button, ButtonProps, Box } from '@mui/material'
import { generatePath, useParams } from 'react-router-dom'
import { Icon } from 'ui/Icons/Icon'
import { useAppBreakpoints } from 'hooks/useAppBreakpoints'
import { InvestRoute } from 'app/pages/invest/router/config'

export const BackToDSOButton = (props: ButtonProps) => {
  const { isMobile } = useAppBreakpoints()
  const { dsoId, issuerId } = useParams<{ dsoId: string; issuerId: string }>()

  return (
    <Box height='100%' display='flex' alignItems='center'>
      <Button
        {...props}
        endIcon={isMobile ? undefined : <Icon name='arrow-right' />}
        startIcon={isMobile ? <Icon name='arrow-left' /> : undefined}
        href={generatePath(InvestRoute.view, { dsoId, issuerId })}
        sx={{
          pl: {
            xs: 0,
            md: 4
          }
        }}
      >
        Back to DSO page
      </Button>
    </Box>
  )
}
