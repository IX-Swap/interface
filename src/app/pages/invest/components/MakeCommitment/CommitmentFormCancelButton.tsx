import React from 'react'
import { Button, ButtonProps } from '@mui/material'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { useParams } from 'react-router-dom'
import { InvestRoute } from 'app/pages/invest/router/config'

export interface CommitmentFormCancelButtonProps extends ButtonProps {}

export const CommitmentFormCancelButton = (
  props: CommitmentFormCancelButtonProps
) => {
  const params = useParams<{ dsoId: string; issuerId: string }>()

  return (
    <Button
      {...props}
      variant='outlined'
      fullWidth
      color='primary'
      disableElevation
      component={AppRouterLinkComponent}
      to={InvestRoute.view}
      params={params}
      replace
    >
      Cancel
    </Button>
  )
}
