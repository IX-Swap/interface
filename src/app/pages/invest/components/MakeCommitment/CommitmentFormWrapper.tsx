import React, { PropsWithChildren } from 'react'
import { CommitmentFormValues } from 'types/commitment'
import { FormProps, Form } from 'components/form/Form'
import { useMakeCommitment } from 'app/pages/invest/hooks/useMakeCommitment'
import {
  commitmentCampaignValidationSchema,
  commitmentNonCampaignValidationSchema
} from 'app/pages/invest/validation'

export interface CommitmentFormProps {
  dso: string
  currency: string
  isCampaign?: boolean
}

export const CommitmentFormWrapper = (
  props: PropsWithChildren<
    CommitmentFormProps & FormProps<CommitmentFormValues>
  >
) => {
  const { dso, currency, children, isCampaign = false, ...rest } = props
  const validationSchema = isCampaign
    ? commitmentCampaignValidationSchema
    : commitmentNonCampaignValidationSchema
  const {
    invest: [makeInvestment]
  } = useMakeCommitment()
  const handleSubmit = async ({
    totalAmount,
    pricePerUnit,
    withdrawalAddress,
    ...values
  }: CommitmentFormValues) => {
    await makeInvestment({
      ...values,
      withdrawalAddress:
        withdrawalAddress === '' ? undefined : withdrawalAddress,
      signedSubscriptionDocument: values.signedSubscriptionDocument?._id,
      dso,
      currency
    })
  }

  return (
    <Form {...rest} onSubmit={handleSubmit} validationSchema={validationSchema}>
      {children}
    </Form>
  )
}
