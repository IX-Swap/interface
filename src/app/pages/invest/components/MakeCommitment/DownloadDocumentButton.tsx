import { DownloadDSOSubscriptionDocument } from 'app/components/DSO/components/DownloadDSOSubscriptionDocument'
import React from 'react'
import { DigitalSecurityOffering } from 'types/dso'
import { Icon } from 'ui/Icons/Icon'

export interface DownloadDocumentButtonProps {
  dso: DigitalSecurityOffering
}

export const DownloadDocumentButton = ({
  dso
}: DownloadDocumentButtonProps) => {
  const isCampaign = dso.isCampaign === true
  const downloadButton = isCampaign
    ? 'Download Investment Agreement'
    : 'Download Subscription Document'

  return (
    <DownloadDSOSubscriptionDocument
      dsoId={dso._id}
      name={dso.subscriptionDocument?.originalFileName}
      fullWidth
      variant='outlined'
      startIcon={<Icon name='download' />}
      sx={{
        borderStyle: 'dashed'
      }}
    >
      {downloadButton}
    </DownloadDSOSubscriptionDocument>
  )
}
