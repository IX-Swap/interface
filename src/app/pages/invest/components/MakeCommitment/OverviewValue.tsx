import { Grid, Typography } from '@mui/material'
import React from 'react'

export interface OverviewValueProps {
  label: string
  value: string
}

export const OverviewValue = ({ label, value }: OverviewValueProps) => {
  return (
    <Grid container spacing={1}>
      <Grid item xs={12}>
        <Typography variant='subtitle1' color='text.secondary'>
          {label}
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <Typography variant='subtitle1'>{value}</Typography>
      </Grid>
    </Grid>
  )
}
