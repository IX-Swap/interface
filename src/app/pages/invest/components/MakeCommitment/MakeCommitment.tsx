import React from 'react'
import { useDSOById } from 'app/pages/invest/hooks/useDSOById'
import { Grid } from '@mui/material'
import { useParams } from 'react-router-dom'
import { useCommitmentActivity } from 'app/pages/invest/hooks/useCommitmentActivity'
import { DSOOverview } from 'app/pages/invest/components/MakeCommitment/DSOOverview'
import { RootContainer } from 'ui/RootContainer'

export const MakeCommitment = () => {
  const params = useParams<{ dsoId: string; issuerId: string }>()
  const { data, isLoading } = useDSOById(params.dsoId, params.issuerId)

  useCommitmentActivity(data?._id)

  if (isLoading || data === undefined) {
    return null
  }

  return (
    <RootContainer>
      <Grid container spacing={3}>
        <Grid item xs={12} md={4}>
          <DSOOverview dso={data} />
        </Grid>
        <Grid item xs={12} md={8}></Grid>
      </Grid>
    </RootContainer>
  )
}
