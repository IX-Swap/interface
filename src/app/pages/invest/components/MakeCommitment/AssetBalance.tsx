import React from 'react'
import { formatMoney } from 'helpers/numbers'
import { OverviewValue } from 'app/pages/invest/components/MakeCommitment/OverviewValue'
import { Grid } from '@mui/material'
import { NoBalance } from 'app/pages/invest/components/MakeCommitment/NoBalance'
import { useCurrencyBalance } from 'app/pages/invest/hooks/useCurrencyBalance'

export interface AssetBalanceProps {
  symbol: string
}

export const AssetBalance = ({ symbol }: AssetBalanceProps) => {
  const currencyBalance: number = useCurrencyBalance(symbol)

  return (
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <OverviewValue
          label='Available Balance'
          value={formatMoney(currencyBalance, symbol)}
        />
      </Grid>
      {currencyBalance <= 0 && (
        <Grid item xs={12}>
          <NoBalance />
        </Grid>
      )}
    </Grid>
  )
}
