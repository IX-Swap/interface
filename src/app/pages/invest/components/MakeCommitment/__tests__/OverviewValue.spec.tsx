import { OverviewValue } from 'app/pages/invest/components/MakeCommitment/OverviewValue'
import React from 'react'
import { render, cleanup } from 'test-utils'

describe('OverviewValue', () => {
  afterEach(async () => {
    await cleanup()
    jest.clearAllMocks()
  })

  it('renders without errors', () => {
    render(<OverviewValue label='Item Label' value='123' />)
  })

  it('should match snapshot', () => {
    const { container } = render(
      <OverviewValue label='Item Label' value='123' />
    )

    expect(container).toMatchSnapshot()
  })
})
