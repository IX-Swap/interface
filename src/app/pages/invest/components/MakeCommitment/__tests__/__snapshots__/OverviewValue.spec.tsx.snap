// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewValue should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-3phns2-MuiTypography-root"
      >
        Item Label
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-1b4l2pg-MuiTypography-root"
      >
        123
      </h6>
    </div>
  </div>
</div>
`;
